docker/Dockerfile.util

```diff
@@ -1,8 +1,8 @@
 FROM openjdk:16-ea-17
 
-RUN apt-get update && \
-  apt-get install -yf postgresql-client netcat && \
-  apt-get clean && \
+RUN microdnf update && \
```

Comment:
did you experience any problems with `apt-get` ?

Reply:
I'm getting:
```
 > [2/4] RUN apt-get update &&   apt-get install -yf postgresql-client netcat &&   apt-get clean &&   rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*:
#6 0.209 /bin/sh: apt-get: command not found
```

Reply:
I was wondering also when I forked `dbt-presto` why is the `openjdk` image used and also why is the postgresql-client needed for checking whether `trino` is up.

With your PR this got me thinking again and I think that it is not really needed.

I've refactored `dbt-trino/docker/Dockerfile.util` in the following fashion:

```
FROM alpine:3.14

RUN apk add --update netcat-openbsd && \
  rm -rf /var/cache/apk/*

COPY wait_for_up.bash /bin/wait_for_up
RUN chmod +x /bin/wait_for_up
```

The file `dbt-trino/docker/wait_for_up.bash` needs also a few small changes to run with `/bin/sh` shell:

```
#!/bin/sh
HOST=$1
PORT=$2
TIMEOUT=${3:-3}
[[ -z $HOST ]] && exit 1
[[ -z $PORT ]] && exit 1

for run in {1..$TIMEOUT}; do
    nc -z $HOST $PORT
    [[ $? -eq 0 ]] && exit 0;
    sleep 10
done

nc -z $HOST $PORT
[[ $? -eq 0 ]] && exit 0;

echo "$HOST:$PORT was not open after $TIMEOUT connection attempts"
exit 1
```

I don't think that we need the heavy image `openjdk` just to do a netcat call.

WDYT ?

Reply:
Yes, it makes sense. I was wondering why `postgresql-client` image is needed, perhaps for connecting to postgresql db from container? Anyway it's not used directly in tests, so let's drop that. Looks like only netcat is used in tests. 

Using `alpine` base image is the right way to go. 

I can include your changes here in my PR.

Reply:
Yes, please.